### PR TITLE
Simplify interrupt module

### PR DESCRIPTION
**Pull request by liuw**
With Vineeth's work committed (https://github.com/cloud-hypervisor/cloud-hypervisor/pull/2414), MSHV and KVM now share a lot more in common.

Take this chance to simplify the interrupt module a bit.

### CODE REVIEW

File: vmm/src/interrupt.rs [hunk 1]
```diff
@@ -115,14 +115,6 @@ impl MsiInterruptGroup<IrqRoutingEntry> {
     }
 }
 
-pub trait RoutingEntryExt {
```
Comment (sboeuf) [inline vmm/src/interrupt.rs:101]: I think keeping the trait is more idiomatic for Rust, and provides an abstraction layer. I don't think we should drop it.

**Review by sboeuf — COMMENTED** [review]
PR looks nice, except that I think the trait `RoutingEntryExt` shouldn't be dropped.
I'll let @rbradford decide about it.

**Review by rbradford — APPROVED** [review]
I think the simplification is helpful in this case.

**Comment by sboeuf** [comment]
Fair enough @rbradford.
Let's merge it!

**Pull request merged**